src/audio/mux/mux.c

```diff
@@ -28,54 +29,58 @@
 #include <stddef.h>
 #include <stdint.h>
 
-static int mux_set_values(struct comp_data *cd, struct sof_mux_config *cfg)
+static int mux_set_values(struct comp_data *cd, struct sof_ipc_stream_map *cfg)
 {
-	uint8_t i;
-	uint8_t j;
-
-	/* check if number of streams configured doesn't exceed maximum */
-	if (cfg->num_streams > MUX_MAX_STREAMS) {
-		trace_mux_error("mux_set_values() error: configured number of "
-				"streams (%u) exceeds maximum = "
-				META_QUOTE(MUX_MAX_STREAMS),
-				cfg->num_streams);
-		return -EINVAL;
+	struct mux_stream_data streams[MUX_MAX_STREAMS];
+	struct sof_ipc_channel_map *ch_map;
+	uint8_t used_streams = 0;
+	uint8_t i, j;
```

Comment:
Btw, any reason for using 8bit maths here ? This can actually evaluate to larger/slower code on RISC processor like xtensa. Can you check this disassembly for any difference to 32 bit, as CC may optimise to 32bit anyway.

Reply:
Just because values are small, changed to 32